packages/babel-core/test/fixtures/transformation/misc/regression-2892/expected.js

```diff
@@ -68,7 +68,7 @@ function _foo() {
       while (1) {
         switch (_context3.prev = _context3.next) {
           case 0:
-            _bar2 = function _bar2() {
+            _bar2 = function _ref2() {
               _bar2 = _asyncToGenerator(
```

Comment:
I think that this was a bug, since `_bar2` wasn't actually reassigned (because the variable was shadowed by the function)